### PR TITLE
feat: Add sample blueprint to run stable diffusion model on inferentia2

### PR DESCRIPTION
**Pull request by ratnopamc**
### What does this PR do?

🛑 Please open an issue first to discuss any significant work and flesh out details/direction - we would hate for your time to be wasted.
Consult the [CONTRIBUTING](https://github.com/awslabs/data-on-eks/blob/main/CONTRIBUTING.md#contributing-via-pull-requests) guide for submitting pull-requests.

This PR showcases a sample blueprint that deploys a stable diffusion model using rayserve on inferentia2 accelerator.
### Motivation
This addresses existing issue - https://github.com/awslabs/data-on-eks/issues/371 

### More

- [x] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [x] Mandatory for new blueprints. Yes, I have added a example to support my blueprint PR
- [x] Mandatory for new blueprints. Yes, I have updated the `website/docs` or `website/blog` section for this feature - work in progress.
- [x] Yes, I ran `pre-commit run -a` with this PR. Link for installing [pre-commit](https://pre-commit.com/) locally

### For Moderators

- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

File: ai-ml/trainium-inferentia/variables.tf [hunk 1]
```diff
@@ -13,7 +13,7 @@ variable "region" {
 
 variable "eks_cluster_version" {
   description = "EKS Cluster version"
-  default     = "1.27"
+  default     = "1.28"
```
Comment (vara-bonthu) [inline ai-ml/trainium-inferentia/variables.tf:16]: you can rebase from the main as this code is already in the main

File: ai-ml/trainium-inferentia/variables.tf [hunk 2]
```diff
@@ -49,3 +49,45 @@ variable "enable_mpi_operator" {
...
+  default = "ON_DEMAND" // change to "SPOT" after testing
+}
+
+variable "inf2_24xl_min_size" {
+  description = "Inf2-24xl Worker node minimum size"
+  type = number
+  default = 0
+}
+
+variable "inf2_24xl_desired_size" {
```
Comment (vara-bonthu) [inline ai-ml/trainium-inferentia/variables.tf:83]: you can rebase from the main as this code is already in the main

File: ai-ml/trainium-inferentia/variables.tf [hunk 3]
```diff
@@ -1,6 +1,6 @@
 variable "name" {
   description = "Name of the VPC and EKS Cluster"
-  default     = "trainium-inferentia"
+  default     = "trainium-inferentia-sd"
```
Comment (vara-bonthu) [inline ai-ml/trainium-inferentia/variables.tf:3]: keep the name as `trainium-inferentia` . this is referred in many places

File: ai-ml/trainium-inferentia/examples/ray-serve/stable-diffusion-inf2/ray-service-stablediffusion.yaml [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+            resources:
+              limits:
+                cpu: "2"
+                memory: "8G"
+              requests:
+                cpu: "2"
+                memory: "8G"
+          nodeSelector:
+            #provisioner: default
+            workload: "rayhead"
```
Comment (vara-bonthu) [inline ai-ml/trainium-inferentia/examples/ray-serve/stable-diffusion-inf2/ray-service-stablediffusion.yaml:61]: Is this using CS or karpenter? you can remove commented config or add details about why its commented

File: ai-ml/trainium-inferentia/eks.tf [hunk 5]
```diff
@@ -486,6 +487,55 @@ module "eks" {
...
+        export PATH=/opt/aws/neuron/bin:$PATH
+      EOT
+
+      min_size     = var.inf2_8xl_min_size
+      max_size     = var.inf2_8xl_max_size
+      desired_size = var.inf2_8xl_desired_size
+
+      labels = {
+        instance-type = "inf2-8xl"
+        //provisioner   = "cluster-autoscaler"
```
Comment (vara-bonthu) [inline ai-ml/trainium-inferentia/eks.tf:518]: remove the commented line

File: ai-ml/trainium-inferentia/eks.tf [hunk 6]
```diff
@@ -498,7 +548,7 @@ module "eks" {
       # aws ssm get-parameters --names /aws/service/eks/optimized-ami/1.27/amazon-linux-2-gpu/recommended/image_id --region us-west-2
       # ami_id   = "ami-0e0deb7ae582f6fe9" # Use this to pass custom AMI ID and ignore ami_type
       ami_type       = "AL2_x86_64_GPU"
-      capacity_type  = "SPOT"
+      capacity_type  = var.inf2_capacity_type
```
Comment (vara-bonthu) [inline ai-ml/trainium-inferentia/eks.tf:551]: Keep it hard-coded this value. Don't create a variable for this

File: ai-ml/trainium-inferentia/eks.tf [hunk 7]
```diff
@@ -507,30 +557,31 @@ module "eks" {
         export PATH=/opt/aws/neuron/bin:$PATH
       EOT
 
-      min_size     = 0
-      max_size     = 2
-      desired_size = 0
+      min_size     = var.inf2_24xl_min_size
+      max_size     = var.inf2_24xl_max_size
```
Comment (vara-bonthu) [inline ai-ml/trainium-inferentia/eks.tf:561]: max_size is hardcoded now in recent changes of the main. pull the latest code from main branch

File: ai-ml/trainium-inferentia/eks.tf [hunk 8]
```diff
@@ -507,30 +557,31 @@ module "eks" {
...
+      desired_size = var.inf2_24xl_desired_size
 
       labels = {
-        instance-type = "inf2"
-        provisioner   = "cluster-autoscaler"
+        instance-type = "inf2-24xl"
+        //provisioner   = "cluster-autoscaler"
```
Comment (vara-bonthu) [inline ai-ml/trainium-inferentia/eks.tf:566]: remove the comment

**Review by vara-bonthu — COMMENTED** [review]
Thanks @ratnopamc ! I have added few comments. Could you also write a Website Doc for end to end deployment?

File: ai-ml/trainium-inferentia/variables.tf (hunk 1 above)
Comment (ratnopamc) [reply ai-ml/trainium-inferentia/variables.tf:16]: Sure, will do.

File: ai-ml/trainium-inferentia/variables.tf (hunk 2 above)
Comment (ratnopamc) [reply ai-ml/trainium-inferentia/variables.tf:83]: Sure, will do.

File: ai-ml/trainium-inferentia/variables.tf (hunk 3 above)
Comment (ratnopamc) [reply ai-ml/trainium-inferentia/variables.tf:3]: Got it.

File: ai-ml/trainium-inferentia/eks.tf (hunk 6 above)
Comment (ratnopamc) [reply ai-ml/trainium-inferentia/eks.tf:551]: Sure, will do.

File: ai-ml/trainium-inferentia/eks.tf (hunk 7 above)
Comment (ratnopamc) [reply ai-ml/trainium-inferentia/eks.tf:561]: Got it.

**Comment by ratnopamc** [comment]
Closed. A new PR https://github.com/awslabs/data-on-eks/pull/406 is tracking this issue.

**Pull request closed without merging**